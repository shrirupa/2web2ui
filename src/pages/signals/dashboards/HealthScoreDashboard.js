import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { list as getSubaccounts } from 'src/actions/subaccounts';
import { getCurrentHealthScore, getInjections } from 'src/actions/signals';
import { Grid } from '@sparkpost/matchbox';
import Page from '../components/SignalsPage';
import HealthScoreOverview from '../containers/HealthScoreOverviewContainer';
import FacetFilter from '../components/filters/FacetFilter';
import DateFilter from '../components/filters/DateFilter';
import SubaccountFilter from '../components/filters/SubaccountFilter';
import CurrentHealthGauge from './components/CurrentHealthGauge/CurrentHealthGauge';
import HealthScoreChart from './components/HealthScoreChart/HealthScoreChart';

export function HealthScoreDashboard(props) {
  const { from, getCurrentHealthScore, getInjections, getSubaccounts, relativeRange, subaccounts, to } = props;

  // Gets subaccount info on mount
  useEffect(() => {
    getSubaccounts();
  }, [getSubaccounts]);

  // Gets injections and current score for gauge and timeseries only when dates change
  useEffect(() => {
    getCurrentHealthScore({ relativeRange, from, to });
    getInjections({ relativeRange, from, to });
  }, [getCurrentHealthScore, getInjections, relativeRange, from, to]);

  return (
    <Page title='Health Score' primaryArea={<DateFilter left />}>
      <Grid>
        <Grid.Column xs={12} lg={5} xl={4}>
          <CurrentHealthGauge />
        </Grid.Column>
        <Grid.Column xs={12} lg={7} xl={8}>
          <HealthScoreChart />
        </Grid.Column>
      </Grid>
      <HealthScoreOverview
        subaccounts={subaccounts}
        hideTitle
        header={
          <Grid>
            <FacetFilter />
            <SubaccountFilter />
          </Grid>
        }
      />
    </Page>
  );
}

const mapStateToProps = (state) => ({
  facet: state.signalOptions.facet,
  from: state.signalOptions.from,
  subaccounts: state.subaccounts.list,
  relativeRange: state.signalOptions.relativeRange,
  to: state.signalOptions.to
});

const mapDispatchToProps = {
  getCurrentHealthScore,
  getSubaccounts,
  getInjections
};

export default connect(mapStateToProps, mapDispatchToProps)(HealthScoreDashboard);
