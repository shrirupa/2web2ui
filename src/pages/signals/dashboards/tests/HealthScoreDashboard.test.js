import { mount, shallow } from 'enzyme';
import React from 'react';
import { HealthScoreDashboard } from '../HealthScoreDashboard';

// Child components are mocked because
// 1. useEffect requires enzyme mount to test
// 2. child components are connected to redux
// This avoids setting up a mock store
jest.mock('../../containers/HealthScoreOverviewContainer');
jest.mock('../components/CurrentHealthGauge/CurrentHealthGauge');
jest.mock('../components/HealthScoreChart/HealthScoreChart');
jest.mock('../../components/filters/FacetFilter');
jest.mock('../../components/filters/DateFilter');
jest.mock('../../components/filters/SubaccountFilter');

describe('Signals Health Score Dashboard', () => {
  const subject = (props = {}, render = shallow) => render(
    <HealthScoreDashboard
      getCurrentHealthScore={() => {}}
      getSubaccounts={() => {}}
      getInjections={() => {}}
      relativeRange='90days'
      from='2015-01-01'
      to='2015-01-05'
      subaccounts={['sub1', 'sub2']}
      {...props}
    />
  );

  it('renders page', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('calls getSubaccounts on mount', () => {
    const getSubaccounts = jest.fn();
    subject({ getSubaccounts }, mount);
    expect(getSubaccounts).toHaveBeenCalled();
  });

  it('calls getCurrentHealthScore getInjections on mount', () => {
    const getInjections = jest.fn();
    const getCurrentHealthScore = jest.fn();
    subject({ getInjections, getCurrentHealthScore }, mount);
    expect(getInjections).toHaveBeenCalledWith({ from: '2015-01-01', relativeRange: '90days', to: '2015-01-05' });
    expect(getCurrentHealthScore).toHaveBeenCalledWith({ from: '2015-01-01', relativeRange: '90days', to: '2015-01-05' });
  });
});
