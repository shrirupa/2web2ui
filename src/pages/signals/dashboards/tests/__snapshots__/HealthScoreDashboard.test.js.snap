// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals Health Score Dashboard renders page 1`] = `
<SignalsPage
  primaryArea={
    <Connect(DateFilter)
      left={true}
    />
  }
  title="Health Score"
>
  <Grid>
    <Grid.Column
      lg={5}
      xl={4}
      xs={12}
    >
      <Connect(CurrentHealthGauge) />
    </Grid.Column>
    <Grid.Column
      lg={7}
      xl={8}
      xs={12}
    >
      <Connect(HealthScoreChart) />
    </Grid.Column>
  </Grid>
  <Connect(withRouter(Connect(HealthScoreOverview)))
    header={
      <Grid>
        <Connect(FacetFilter) />
        <Connect(Connect(SubaccountFilter)) />
      </Grid>
    }
    hideTitle={true}
    subaccounts={
      Array [
        "sub1",
        "sub2",
      ]
    }
  />
</SignalsPage>
`;
